service/r2/routes.go

```diff
@@ -271,3 +271,27 @@ func fetchRollupRuleHistory(s *service, r *http.Request) (data interface{}, err
 	}
 	return models.NewRollupRuleSnapshots(hist), nil
 }
+
+func updateRuleSet(s *service, r *http.Request) (data interface{}, err error) {
+	var req updateRuleSetRequest
+	if err := parseRequest(&req, r.Body); err != nil {
+		return nil, NewBadInputError(err.Error())
+	}
+	uOpts, err := s.newUpdateOptions(r)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(req.RuleSetChanges.MappingRuleChanges) == 0 &&
```

Comment:
Check this immediately after `parseRequest`? Usually better to perform as much validation as possible upfront.